platform/src/main/java/org/stellar/anchor/platform/paymentobserver/CirclePaymentObserverService.java

```diff
@@ -0,0 +1,217 @@
+          .setPrettyPrinting()
+          .create();
+
+  public CirclePaymentObserverService(
+      OkHttpClient httpClient, CirclePaymentObserverConfig circlePaymentObserverConfig) {
+    this.httpClient = httpClient;
+    this.circlePaymentObserverConfig = circlePaymentObserverConfig;
+    Network stellarNetwork = StellarNetworkHelper.toStellarNetwork(circlePaymentObserverConfig.getStellarNetwork());
+    String[] assetIdPieces = CircleAsset.stellarUSDC(stellarNetwork).split(":");
+    this.usdcIssuer = assetIdPieces[assetIdPieces.length - 1];
+  }
+
+  public void handleCircleNotification(Map<String, Object> requestBody) {
+    CircleNotification circleNotification =
+        gson.fromJson(gson.toJson(requestBody), CircleNotification.class);
+    String type = circleNotification.getType();
+
+    switch (type) {
+      case "SubscriptionConfirmation":
+        handleSubscriptionConfirmationNotification(circleNotification);
+        return;
+
+      case "Notification":
+        handleTransferNotification(circleNotification);
+        return;
+
+      default:
+        Log.warn("Not handling notification of type " + type);
+    }
+  }
+
+  /**
+   * This will auto-subscribe to Circle when we receive a subscription available notification.
+   * @param circleNotification is the circle notification object.
+   */
+  public void handleSubscriptionConfirmationNotification(CircleNotification circleNotification) {
+    String subscribeUrl = circleNotification.getSubscribeUrl();
+    Log.info("=====> subscriptionNotification.subscribeUrl: " + subscribeUrl);
+
+    Request httpRequest =
+        new Request.Builder()
+            .url(subscribeUrl)
+            .header("Content-Type", "application/json")
+            .get()
+            .build();
+    Response response;
+    try {
+      response = httpClient.newCall(httpRequest).execute();
+    } catch (IOException e) {
+      Log.error("Failed to call endpoint " + subscribeUrl);
+      return;
+    }
+
+    Log.info(
+        String.format(
+            "Called subscribeUrl %s and got status code %d", subscribeUrl, response.code()));
+    if (response.code() != 200) {
+      Log.error(String.format("Status code: %d. \nResponse: %s", response.code(), response.body()));
+    }
+  }
+
+  /**
+   * Handle incoming circle notifications of type "transfers". A transfer notification can contain circle<>circle or circle<>stellar events.
+   * @param circleNotification is the circle notification object.
+   */
+  public void handleTransferNotification(CircleNotification circleNotification) {
+    TransferNotificationBody transferNotification =
+        gson.fromJson(circleNotification.getMessage(), TransferNotificationBody.class);
+
+    CircleTransfer circleTransfer = transferNotification.getTransfer();
+    if (circleTransfer == null) {
+      Log.error("Missing \"transfer\" value in notification of type \"transfers\".");
```

Comment:
This can be treated the same way as line#113. I would suggest that we use Log.info